src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -2568,7 +2570,7 @@ private void PopulateAssemblyRefTableRows()
             foreach (var assemblyRef in assemblyRefs)
             {
                 AssemblyRefTableRow r = new AssemblyRefTableRow();
-                var identity = assemblyRef.Identity;
+                var identity = MapAssemblyReferenceIdentity(assemblyRef.Identity);
```

Comment:
Is it possible that we emit duplicate rows because of this remapping? It would be good to at least assert that this doesn't happen.